docs/index.md

```diff
@@ -459,6 +459,531 @@ class Settings(BaseSettings):
+
+    v0: str
+    """the top level v0 option"""
+
+    sub_model: SubModel
+    """The help summary for SubModel related options. This will be placed at top of group."""
+
+
+Settings(_cli_prog_name='app', _cli_parse_args=['--help'])  # (1)!
+"""
+usage: app [-h] [--v0 str] [--sub_model JSON] [--sub_model.v1 int] [--sub_model.deep JSON]
+           [--sub_model.deep.v4 List[int]]
+
+The Settings class documentation will show in top level help text.  # (2)!
+
+options:
+  -h, --help            show this help message and exit
+  --v0 str              the top level v0 option  # (3)!
+
+sub_model options:  # (4)!
+  The help summary for SubModel related options. This will be placed at top of group.
+
+  --sub_model JSON      set sub_model from JSON string
+  --sub_model.v1 int    the sub model v1 option   # (5)!
+
+sub_model.deep options:
+  The help summary for DeepSubModel and related options. This will be placed at top of
+  group.  # (6)!
+
+  --sub_model.deep JSON  # (7)!
+                        set sub_model.deep from JSON string
+  --sub_model.deep.v4 List[int]
+                        the deeply nested sub model v4 option
+"""
+```
+
+1. Does `_cli_prog_name` and `_cli_parse_args` look familiar? They retain the same meanings as in argparse.
+
+2. Help text for application main or subcommands is populated from class docstrings.
+
+3. Help text for fields is populated from field descriptions.
+
+4. Nested models (e.g. `SubModel`, `DeepSubModel`) and their associated fields will always be grouped together.
+
+5. Note that nested fields look and act just like their environment variable counterparts. The CLI uses `.` as its
+   nested delimiter.
+
+6. Group help text is populated from field descriptions by default, but can be configured to pull from class docstrings
+   as well.
+
+7. Just like when parsing environment variables, top level models allow for JSON strings and nested fields taking
+   precedence.
+
+To enable CLI parsing, we simply set the `cli_parse_args` flag to a valid value, which retains similar conotations as
+defined in argparse. In the above example, we parsed our args from the `['--help']` list that was passed into
+`_cli_parse_args`. Alternatively, we could have set `_cli_parse_args=True` to parse args from the command line (i.e.,
+`sys.argv[1:]`).
+
+Lastly, a CLI settings source is always [**the topmost source**](#field-value-priority), and does not support [changing
+its priority](#changing-priority).
+
+#### Enable CLI Argument Parsing
+
+`cli_parse_args: Optional[Union[List[str], bool]] = None`
+
+* Default = `None`
+* If `True`, parse from `sys.argv[1:]`
+* If `List[str]`, parse from `List[str]`
+* If `False` or `None`, do not parse CLI arguments
+
+#### Lists
+
+CLI argument parsing of lists supports intermixing of any of the below three styles:
+
+  * JSON style `--field='[1,2]'`
+  * Argparse style `--field 1 --field 2`
+  * Lazy style `--field=1,2`
+
```

Comment:
3 seems excessive, surely just Argparse and lazy should be allowed.

Or Argparse and JSON if we need json elsewhere?

Reply:
I would lean towards JSON and Argparse, mostly to maintain consistency with JSON input and environment variable parsing. Of course, it's annoying to write `[...]` any time you need a list at the CLI, but it's pretty minor. 

However, from an implementation point of view we would have to go out of our way to exclude one of the styles. It's actually easier to support all three. The list parsing work was really dictated by the dict key=val list parsing. If we removed that, we could simplify the list parsing implementation.

Reply:
I left all three options in as it is simplest given we support dict key=val list parsing.